cristin-project/src/main/java/no/unit/nva/cristin/projects/update/UpdateProjectResourceAccessCheck.java

```diff
@@ -24,24 +24,25 @@ public class UpdateProjectResourceAccessCheck extends ProjectResourceAccessCheck
 
     private transient boolean verified;
 
+    
     @Override
     public void verifyAccess(NvaProject resource, Map<String, String> params) {
         var userIdentifier = params.get(USER_IDENTIFIER);
 
-        var resourceCreator = getResourceCreator(resource);
-        var resourceManager = getResourceManager(resource);
-
-        if (resourceCreator.isPresent() && hasMatch(resourceCreator.get(), userIdentifier)) {
-            verified = true;
+        getResourceCreator(resource)
+            .ifPresentOrElse(creator -> verified = hasMatch(creator, userIdentifier),
+                             () -> verified = false);
+        if (verified) {
             return;
-        } else if (resourceManager.isPresent() && hasMatch(resourceManager.get(), userIdentifier)) {
-            verified = true;
+        }
+
+        getResourceManager(resource)
+            .ifPresent(manager -> verified = hasMatch(manager, userIdentifier));
```

Comment:
or atleast use an` ifPresentOrElse` here as well and do` verified = false` on ifElse

Reply:
or if im thinking correclty, the hasMatch here will do the final setting of the variable verified, so we dont need it further down?

Reply:
on line 32  I do 
```
 getResourceCreator(resource)
            .ifPresentOrElse(creator -> verified = hasMatch(creator, userIdentifier),
                             () -> verified = false);
```
this will ensure that verified has a value.

on line 39 we only need to validate  a match,  because verified has already been set to false.